### PR TITLE
fix: Add adminReview option to allow impersonating org members

### LINKED ISSUE #14276: [CAL-3336] Introduce admin Reviewed option for an organization
An instance ADMIN should be able to mark an organization as reviewed which would then allow organization admins to impersonate it's members.

This is necessary so that auto-added users can't be impersonated by a self-serve created organization, the owner of whom could be a data thief or hacker.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3336](https://linear.app/calcom/issue/CAL-3336/introduce-admin-reviewed-option-for-an-organization)</sub>

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Fixes #14276
[Loom Demo](https://www.loom.com/share/fbc572511c6f493c88c2d5fde2d76cea)

- Marks all existing organizations as adminReviewed
- All orgs created by ADMIN would be automatically marked as adminReviewed
- An org created by a regular user can be marked adminReviewed from admin interface.

## Type of change
- Bug fix (non-breaking change which fixes an issue)

### Self-Serve Test
1. Create an organization through self-serve flow. Test that members can't be impersonated yet
2. Login  as instance ADMIN in a separate browser instance and mark the organization as reviewed. Notice that now the members can be impersonated.
### Instance Admin creation Test
1. Create an organization after logging in as admin.
2. Now login as the organization owner and notice that impersonation is allowed straightaway without manual review required.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist
- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by hariombalhara** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/calcom/cal.com/14275?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#14275** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/14275?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#13139** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/13139?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hariombalhara and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/features/ee/organizations/pages/settings/admin/AdminOrgPage.tsx [hunk 1]
```diff
@@ -80,14 +80,13 @@ function AdminOrgTable() {
         <Header>
           <ColumnTitle widthClassNames="w-auto">{t("organization")}</ColumnTitle>
           <ColumnTitle widthClassNames="w-auto">{t("owner")}</ColumnTitle>
-          <ColumnTitle widthClassNames="w-auto">{t("verified")}</ColumnTitle>
```
Comment (hariombalhara) [inline packages/features/ee/organizations/pages/settings/admin/AdminOrgPage.tsx:83]: Removed verified as all organizations are auto-verified whether created using self-serve or by admin.

File: packages/prisma/schema.prisma [hunk 2]
```diff
@@ -392,9 +392,13 @@ model OrganizationSettings {
   organization                  Team       @relation(fields: [organizationId], references: [id], onDelete: Cascade)
   organizationId                Int        @unique
   isOrganizationConfigured      Boolean    @default(false)
+  // It decides if new organization members can be auto-accepted or not
   isOrganizationVerified        Boolean    @default(false)
   orgAutoAcceptEmail            String
   lockEventTypeCreationForUsers Boolean    @default(false)
+  // It decides if instance ADMIN has reviewed the organization or not. 
+  // It is used to allow super sensitive operations like 'impersonation of Org members by Org admin'
+  isAdminReviewed               Boolean    @default(false)
```
Comment (hariombalhara) [inline packages/prisma/schema.prisma:401]: Using the term reviewed to clearly differentiate from existing verified flag.

File: packages/features/users/components/UserTable/UserListTable.tsx [hunk 3]
```diff
@@ -127,9 +127,8 @@ export function UserListTable() {
         placeholderData: keepPreviousData,
       }
     );
-
   const totalDBRowCount = data?.pages?.[0]?.meta?.totalRowCount ?? 0;
-  const adminOrOwner = currentMembership?.user.role === "ADMIN" || currentMembership?.user.role === "OWNER";
+  const adminOrOwner = org?.user.role === "ADMIN" || org?.user.role === "OWNER";
```
Comment (hariombalhara) [inline packages/features/users/components/UserTable/UserListTable.tsx:131]: `org` is better name as it isn't just membership

File: packages/features/users/components/UserTable/UserListTable.tsx [hunk 4]
```diff
@@ -109,7 +109,7 @@ function reducer(state: State, action: Action): State {
 
 export function UserListTable() {
   const { data: session } = useSession();
-  const { data: currentMembership } = trpc.viewer.organizations.listCurrent.useQuery();
+  const { data: org } = trpc.viewer.organizations.listCurrent.useQuery();
```
Comment (hariombalhara) [inline packages/features/users/components/UserTable/UserListTable.tsx:112]: Renamed to org because `org` is better name as it isn't just membership

**Review by sean-brydon — COMMENTED** [review]
Code LGTM - but can we not reuse isOrganizationVerified - i think that was the whole point in this field was for this. I had a quick look around and can't see it being used for much outside of this usecase

**Review by sean-brydon — COMMENTED** [review]
As discussed in DMs - LGTM.

I'll wait for the main org self serve flow to be merged first before we merge this

**Review by sean-brydon — APPROVED** [review]
LGTM - nice job!

**Pull request merged**